modules/mod_ginger_base/support/ginger_geo_search.erl

```diff
@@ -19,45 +19,49 @@ search_query(#search_query{search={ginger_geo, Args}}, Context) ->
     };
 
 %% @doc Similar to z_geo_search:search_query/2 but result set includes locations and category
-search_query(#search_query{search={ginger_geo_nearby, Args}}, Context) ->
-
+search_query(#search_query{search = {ginger_geo_nearby, Args}}, Context) ->
     BaseArgs = lists:filter(
         fun({Key, _}) ->
             not lists:member(Key, [distance, lat, lng])
         end,
         Args
     ),
-    BaseSearch = ginger_geo_search:search_query(#search_query{search={ginger_geo, BaseArgs}}, Context),
+    BaseSearch = search_query(#search_query{search = {ginger_geo, BaseArgs}}, Context),
+    with_geo_nearby(BaseSearch, Args, Context).
 
-    Distance = z_convert:to_float(proplists:get_value(distance, Args, 10)),
-    {Lat, Lng} = z_geo_search:get_query_center(Args, Context),
-    {LatMin, LngMin, LatMax, LngMax} = z_geo_support:get_lat_lng_bounds(Lat, Lng, Distance),
+%% @doc Extend an SQL query with geo nearby search based on Args.
+-spec with_geo_nearby(#search_sql{}, proplists:proplist(), z:context()) -> #search_sql{}.
+with_geo_nearby(#search_sql{} = SearchSql, Args, Context) ->
+    GeoArgs = geo_args(Args, Context),
+    {Placeholders, SearchWithGeoArgs} = add_args(GeoArgs, SearchSql),
 
-    WhereStr = string:join(
-        [
-            z_convert:to_list(LatMin),
-            "< rsc.pivot_location_lat AND",
-            z_convert:to_list(LngMin),
-            "< rsc.pivot_location_lng AND",
-            "rsc.pivot_location_lat <",
-            z_convert:to_list(LatMax),
-            "AND rsc.pivot_location_lng <",
-            z_convert:to_list(LngMax)
-        ],
-        " "
-    ),
-    SearchSql = and_where(BaseSearch, WhereStr),
+    %% Rebuild WHERE and ORDER BY because argument positions have shifted.
+    GeoWhere = arg(1, Placeholders) ++ " < pivot_location_lat AND " ++ arg(2, Placeholders) ++ "< pivot_location_lng "
+            ++ " AND pivot_location_lat < " ++ arg(3, Placeholders) ++ " AND pivot_location_lng < " ++ arg(4, Placeholders),
+    GeoOrder = "(pivot_location_lat-" ++ arg(5, Placeholders) ++ ")^2"
+        ++ " + (pivot_location_lng-" ++ arg(6, Placeholders) ++ ")^2",
```

Comment:
This solves the actual bug: by keeping the coordinate floats as query parameters (instead of merging them into one query string), escaping any negative floats works properly.

Reply:
The original query has `A*A` but `A^2` is of course the same and doesn’t need the argument repetition.